src/Framework/Models/EagerLoader.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Give\Framework\Models;
+
+use ReflectionClass;
+
+/**
+ * @unreleased
+ */
+class EagerLoader
+{
+    /**
+     * @unreleased
+     * @var string
+     */
+    protected $modelClass;
+
+    /**
+     * @unreleased
+     * @var string
+     */
+    protected $relationshipKey;
+
+    /**
+     * @unreleased
+     * @var ModelQueryBuilder
+     */
+    protected $modelQuery;
+
+    /**
+     * @unreleased
+     * @var ModelQueryBuilder
+     */
+    protected $eagerLoadedQuery;
+
+    /**
+     * @unreleased
+     * @var string
+     */
+    protected $foreignKey;
+
+    /**
+     * @var mixed
+     */
+    protected $foreignAttribute;
+
+    public function __construct($modelClass, $relationshipKey, $eagerLoadedModelClass, $foreignKey, $foreignAttribute = null)
```

Comment:
Let's add types to these params and a docblock for the construct.

Reply:
I think I figured out the class string typing.

Reply:
Resolved, see https://github.com/impress-org/givewp/pull/7227/commits/69707995eddb9f276cc69aa9c51272f72cc207e4.